File: Examples/ObjC/RBQFetchedResultsControllerExample/TestObject.m
//
//  TestObject.m
//  RBQFetchedResultsControllerTest
//
//  Created by Lauren Smith on 1/2/15.
//  Copyright (c) 2015 Roobiq. All rights reserved.
//

#import "TestObject.h"

@implementation TestObject

+ (NSString *)primaryKey
{
    return @"key";
}

// Specify default values for properties

//+ (NSDictionary *)defaultPropertyValues
//{
//    return @{};
//}

// Specify properties to ignore (Realm won't persist these)

//+ (NSArray *)ignoredProperties
//{
//    return @[];
//}

+ (instancetype)testObjectWithTitle:(NSString *)title
                          sortIndex:(NSInteger)sortIndex
                            inTable:(BOOL)inTable
{
    TestObject *object = [[TestObject alloc] init];
    object.sortIndex = sortIndex;
    object.title = title;
    object.key = [NSString stringWithFormat:@"%@%ld",title, (long)sortIndex];
    object.inTable = inTable;
    
    return object;
}

- (id)copyWithZone:(nullable NSZone *)zone
{
    TestObject *object = [[TestObject allocWithZone:zone] init];
    object.title = self.title;
    object.sortIndex = self.sortIndex;
    object.sectionName = self.sectionName;
    object.key = self.key;
    object.inTable = self.inTable;
    
    return object;
}

@end

File: RBQFetchedResultsControllerSwiftExample/RBQFetchedResultsControllerSwiftExample/TestObject.swift
//
//  TestObject.swift
//  TimeHenPersonal
//
//  Created by Jeff on 4/1/15.
//  Copyright (c) 2015 Jeff. All rights reserved.
//


import Realm

class TestObject: RLMObject {
    dynamic var title : NSString = "Default Title"
    dynamic var sortIndex : NSInteger = 0
    dynamic var sectionName : NSString = "Section 1"
    dynamic var key : NSString = "Section 1"
    dynamic var inTable : Bool = true
    
    override class func primaryKey() -> String {
        return "key"
    }
    
//    init(title: NSString, sortIndex: NSInteger, inTable: Bool) {
//        super.init()
//        self.sortIndex = sortIndex
//        self.title = title
//        self.key = NSString(format: "%@%ld", title, CLong(sortIndex))
//        self.inTable = inTable
//    }
}
